live-build/misc/upgrade-scripts/upgrade-container

```diff
@@ -365,25 +378,73 @@ function convert_to_bootfs() {
 	migrate_password_for_user root
 
 	#
-	# LX-72 Until we have a proper solution for migrating the
-	# configuration of a given appliance to the new root filesystem
-	# (i.e. any configuration that isn't supplied by a package), we
-	# resort to explicitly migrating some of this configuration
-	# here. The files listed here are dynamically generated and/or
+	# These files are generic OS files that are required for the
+	# system to properly operate; these must be preserved acorss
+	# upgrades.
+	#
+	while read -r file; do
+		migrate_file "$file"
+	done <<-EOF
+		/etc/hostname
+		/etc/hosts
+		/etc/machine-id
+		/etc/resolv.conf
```

Comment:
On my DE system this is a symlink to `./run/systemd/resolve/resolv.conf`. Does `migrate_file` handle symlinks? I See you've added `-a` to cp for `migrate_directory`, perhaps you need something similar for `migrate_file`?

Reply:
it should probably be commented better, but I think we actually don't want to migrate this file if it's the symlink (which is the default).

it won't be a symlink if the app-gate has modified it, to use some custom DNS settings, and that's when we need to do the migration.

otherwise, leaving it as not migrated will enable it to use the default systemd-resolved behavior. since we do the `-f` test in `migrate_file`, I think it won't copy anything when it's a symlink.

Reply:
I definitely think you should definitely add a comment. This holds in the case of `/etc/resolv.conf`, but might not necessarily hold for other configuration files/links, so it'd be important for that behaviour to be very obvious.